sphinx_gallery/gen_gallery.py

```diff
@@ -181,24 +181,23 @@ def generate_gallery_rst(app):
         computation_times += this_computation_times
 
         # we create an index.rst with all examples
-        fhindex = open(os.path.join(gallery_dir, 'index.rst'), 'w')
-        # :orphan: to suppress "not included in TOCTREE" sphinx warnings
-        fhindex.write(":orphan:\n\n" + this_fhindex)
-        for directory in sorted(os.listdir(examples_dir)):
-            if os.path.isdir(os.path.join(examples_dir, directory)):
-                src_dir = os.path.join(examples_dir, directory)
-                target_dir = os.path.join(gallery_dir, directory)
-                this_fhindex, this_computation_times = generate_dir_rst(src_dir, target_dir, gallery_conf,
-                                                                        seen_backrefs)
-                fhindex.write(this_fhindex)
-                computation_times += this_computation_times
-
-        if gallery_conf['download_all_examples']:
-            download_fhindex = generate_zipfiles(gallery_dir)
-            fhindex.write(download_fhindex)
-
-        fhindex.write(SPHX_GLR_SIG)
-        fhindex.flush()
+        with open(os.path.join(gallery_dir, 'index.rst'), 'wb') as fhindex:
```

Comment:
I prefer codecs.open with encoding='utf-8', then forget about the encode instruction down the line